src/ray/core_worker/core_worker.cc

```diff
@@ -3773,16 +3804,24 @@ void CoreWorker::HandleGetObjectLocationsOwner(
     rpc::GetObjectLocationsOwnerRequest request,
     rpc::GetObjectLocationsOwnerReply *reply,
     rpc::SendReplyCallback send_reply_callback) {
-  auto &object_location_request = request.object_location_request();
-  if (HandleWrongRecipient(
-          WorkerID::FromBinary(object_location_request.intended_worker_id()),
-          send_reply_callback)) {
+  if (HandleWrongRecipient(WorkerID::FromBinary(request.intended_worker_id()),
+                           send_reply_callback)) {
     return;
   }
-  auto object_id = ObjectID::FromBinary(object_location_request.object_id());
-  auto object_info = reply->mutable_object_location_info();
-  auto status = reference_counter_->FillObjectInformation(object_id, object_info);
-  send_reply_callback(status, nullptr, nullptr);
+  Status overall_status = Status::OK();  // Assume success initially
+  for (int i = 0; i < request.object_id_size(); ++i) {
+    auto object_id = ObjectID::FromBinary(request.object_id(i));
+    auto object_info = reply->add_object_location_infos();
+    // TODO(zhilong): Need to deal with fail query for each object?
+    auto current_status =
+        reference_counter_->FillObjectInformation(object_id, object_info);
```

Comment:
I checked `FillObjectInformation()`, it always return Status::OK(). Could you change the method signature to return void so the caller doesn't need to handle error.

Reply:
I just change FillObjectInformation from Status to OK, and also make some modifications to the test